core/src/result.rs

```diff
@@ -31,6 +31,12 @@ pub enum Error {
     #[error("parsing failed: {0}")]
     Parser(String),
 
+    #[error("OverflowError: {0}")]
+    OverflowError(String),
```

Comment:
I think it can be improved by using Operator Enum, Lhs, and Rhs.

Reply:
Ok.. will do.

Reply:
thanks for your hard work...

Reply:
When I move the error message to the ValueError enum, I get the following error:
 _ => Err(ValueError::F64ToDecimalConversionError(lhs)),
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `result::Error`, found enum `ValueError` 

I have gotten this before, and I cannot figure out why Err doesn't want to accept a ValueError sometimes.

Reply:
What the function returns is the top-level `Error` enum.
But where we declared the error is in the `ValueError` enum.
If so, here's a very nice way to convert from ValueError to top-level Error.
That's using the `Into` trait.

Since we already use the Error macro + `#from` function of `thiserror`, it is already implemented.

try it like this
`Err(ValueError::F64ToDecimalConversionError(lhs).into())`

Reply:
Thanks..  I was wondering why into was used in the error messages..